### ISSUE TITLE
Unable to install jre8 from github repository

### ISSUE THREAD

**Issue by travispaul** [saltstack/salt#31408]
As instructed in the [jre8 package definition](https://github.com/saltstack/salt-winrepo-ng/blob/master/jre8.sls#L7), I have downloaded the jre-8u73-windows-x64.exe to `salt://win/repo-ng/jre8/` and executed `pkg.refresh_db`.  I can install other packages ([npp ](https://github.com/saltstack/salt-winrepo-ng/blob/master/npp.sls)for example) but not jre8.  

I receive the following error:

```
----------
          ID: jre8
    Function: pkg.installed
      Result: False
     Comment: The following packages failed to install/update: jre8=8.0.730.2
     Started: 08:14:38.034000
    Duration: 189281.0 ms
     Changes:
              ----------
              _comment:
                  Registry not updated.
```

my init.sls:

``` YAML
jre8:
  pkg:
    - installed
```

pkg.available_version:

```
[root@salt salt]# salt MINION pkg.available_version jre8
MINION:
    8.0.730.2
```

version report:

```
Salt Version:
           Salt: 2015.8.5

Dependency Versions:
         Jinja2: unknown
       M2Crypto: 0.20.2
           Mako: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.3.1
         Python: 2.6.6 (r266:84292, Jul 23 2015, 15:22:56)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 3.2.5
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.8.1
        timelib: Not Installed

System Versions:
           dist: centos 6.7 Final
        machine: x86_64
        release: 2.6.32
         system: CentOS 6.7 Final
```

Minion is 2015.8.5

**Comment by jfindlay** [comment]
@TravisPaul, thanks for reporting.  @thebigbear, do you have any ideas on this one?

**Comment by TheBigBear** [comment]
@jfindlay no, I don't. @TravisPaul  what happens if you do an explicit 

```'salt 'your-minion-id' pkg.install jre8'```, 

instead of doing it via a state? Just trying to narrow it down.

**Comment by TravisPaul** [comment]
@TheBigBear I get the same results with ```salt MINION pkg.install jre8``` and ```salt MINION pkg.install jre8 version=8.0.730.2```

```YAML
windev1.weblinkid.com:
    ----------
    _comment:
        Registry not updated.
```

**Comment by TheBigBear** [comment]
@TravisPaul and jre 8 is definetely not installed on that windev1.weblinkid.com minion?
I think "Registry not updated" is also the response you get if jre 8 is already on the minion.

**Comment by TravisPaul** [comment]
@TheBigBear correct, the only thing I've installed on the machines so far has been salt (and Notepadd++ from the pkg-ng repo to test).

![screen shot 2016-02-23 at 5 42 47 pm](https://cloud.githubusercontent.com/assets/2118993/13269284/f9057a8e-da54-11e5-8256-584d72b9f48e.png)

**Comment by TravisPaul** [comment]
Also, I just installed the jre8 manually on that server, and it returns the same message:

```
[root@salt ~]# salt MINION pkg.install jre8
MINION:
    ----------
    _comment:
        Registry not updated.
```

And it is shown as installed:
```
[root@salt ~]# salt MINION pkg.list_pkgs   
MINION:
    ----------
    Configuration Manager Client:
        5.00.8239.1000
    Java 8 Update 73:
        8.0.730.2
    Java Auto Updater:
        2.8.73.2
    Microsoft Endpoint Protection Management Components:
        4.7.0214.0
    Microsoft Forefront Endpoint Protection 2010 Server Management:
        4.7.0214.0
    Microsoft Policy Platform:
        1.2.3602.0
    Microsoft Security Client:
        4.7.0214.0
    Microsoft Visual C++ 2013 Redistributable (x64) - 12.0.21005:
        12.0.21005.1
    Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.21005:
        12.0.21005.1
    Microsoft Visual C++ 2013 x64 Additional Runtime - 12.0.21005:
        12.0.21005
    Microsoft Visual C++ 2013 x64 Minimum Runtime - 12.0.21005:
        12.0.21005
    Microsoft Visual C++ 2013 x86 Additional Runtime - 12.0.21005:
        12.0.21005
    Microsoft Visual C++ 2013 x86 Minimum Runtime - 12.0.21005:
        12.0.21005
    System Center Endpoint Protection:
        4.7.214.0
    Windows Firewall Configuration Provider:
        1.2.3412.0
    ms-vcpp-2005-sp1-redist_x64:
        8.0.61000
    npp:
        6.8.8
    saltstack.minion:
        2015.8.5
    silverlight:
        5.1.30514.0
```

**Comment by TheBigBear** [comment]
@TravisPaul thanks, that was my point. As I had suspected it is actually installed and when it is, that is what you get. 

Please un-install it manually and then install it using salt and I exepct you find it will work. 

I suspect there is possibly a bug/misinterpretation inside salt of what happens and how salt interprets what signals it sends internally when it is already present. @jfindlay can you look into this angle once @TravisPaul confirms that it works OK, when it is NOT there to start with?

**Comment by TravisPaul** [comment]
@TheBigBear I installed it **MANUALLY**, that's how I got that result. Uninstalling and reinstalling does not result any any change.  I have a couple of other newly installed VMs with the same outcome.

**Comment by TheBigBear** [comment]
@TravisPaul what do you mena when you say "manually"? When I say "manually" I think of issuing the install comand manually on the cmd line, not using "salt pkg.install jre8".

I am asking you to remove it "manually" by going into control panel and remove the software using standard windows out of thr box uninstall. 

And after that using 'salt pkg.install jre8' and I think that will work? After that use it a second time and I expect it will return the error you are showing above?

**Comment by TravisPaul** [comment]
By manulally, I mean that I logged into the machine, downloaded and installed the jre entirely outside of salt.  Then, using salt, I attempted to install and received ```Registry not updated.```.  If I uninstall jre from windows using Add or Remover Programs, then try to install with salt.  I still receive ```Registry not updated.```.

I tried this earlier but I also just went through that exact process again to ensure I didn't miss anything.  To sum up I get ```Registry not updated.``` when trying to install though salt, when the jre is installed and also when its not installed.
